docs/sources/flow/reference/components/otelcol.auth.sigv4.md

```diff
@@ -0,0 +1,171 @@
+---
+title: otelcol.auth.sigv4
+---
+
+# otelcol.auth.sigv4
+
+`otelcol.auth.sigv4` exposes a `handler` that can be used by other `otelcol`
+components to authenticate requests to AWS services using the AWS Signature Version 4 (SigV4) protocol. 
+See [here](https://docs.aws.amazon.com/general/latest/gr/signing-aws-api-requests.html) for more
+information about SigV4.
+
+> **NOTE**: `otelcol.auth.sigv4` is a wrapper over the upstream OpenTelemetry
+> Collector `sigv4auth` extension. Bug reports or feature requests will be
+> redirected to the upstream repository, if necessary.
+
+Multiple `otelcol.auth.sigv4` components can be specified by giving them
+different labels.
+
+> **NOTE**: The Agent must have valid AWS credentials as used by the 
+[AWS SDK for Go](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials).
+
+## Usage
+
+If the exporter URL does not contain enough information about the region and service, 
+the minimum arguments are:
+```river
+otelcol.auth.sigv4 "LABEL" {
+  region  = "REGION"
+  service = "SERVICE"
+}
+```
+
+In some cases the exporter URL does contain enough information about the region and service as 
+described [here](#arguments). In that case there is no need to specify any arguments to this component:
+```river
+otelcol.auth.sigv4 "LABEL" {
+}
+```
+
+## Arguments
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`region` | `string` | The AWS region for the service you are exporting to for AWS SigV4. | "" | no
+`service` | `string` | The AWS service for AWS SigV4. | "" | no
+
+Both `region` and `service` are required for the component to work, but if they are left empty their values 
+may be inferred from the exporter URL based on 
```

Comment:
They're not required, then :) I think this needs some rephrasing to talk about how they must be provided if the URL in the exporters don't follow a set of rules.

Reply:
Makes sense. What do you think about this instead:

> `region` and `service` only need to be specified if their values cannot be inferred from the exporter URL.
If they are left empty, their values may be inferred from the exporter URL based on the following rules:

Reply:
Maybe flip it around?

> If `region` and `service` are left empty, their values are inferred from the URL of the exporter using this component using the following rules:
>
> (RULES)
>
> If none of the above rules apply, then `region` and `service` must be specified.